extmod/modlwip.c

```diff
@@ -1132,11 +1132,11 @@ STATIC mp_uint_t lwip_socket_ioctl(mp_obj_t self_in, mp_uint_t request, uintptr_
             ret |= MP_STREAM_POLL_RD;
         }
 
-        if (flags & MP_STREAM_POLL_WR && tcp_sndbuf(socket->pcb.tcp) > 0) {
+        if (flags & MP_STREAM_POLL_WR && socket->pcb.tcp != NULL && tcp_sndbuf(socket->pcb.tcp) > 0) {
             ret |= MP_STREAM_POLL_WR;
         }
 
-        if (socket->state == STATE_PEER_CLOSED) {
+        if (socket->state == STATE_PEER_CLOSED || socket->pcb.tcp == NULL) {
```

Comment:
Can you please explain the logic behind this change?  If pcb.tcp==NULL and one tries to write to the socket then it will seg fault right away.

Reply:
Inspired by your comment in #3164
> The issue with crashing when polling a closed/invalid socket is due to lwip_socket_ioctl dereferencing the null pointer socket->pcb.tcp. It's easy to add a check to guard against this....

and having this issue constantly reproducible on `esp8266` - suggested fix at least works.

Without deep knowledge how `lwip` works, I can make only assumptions:

#### Setup:

1. Having open/alive TCP connection
2. Having server constantly sending data (e.g. filetransfer)
3. After some time client unexpectedly closes connection.

#### Usual workflow for server:
1. Wait until socket write buffer has some free space (`select.poll(POLL_WR)`)
2. Send next portion of data

#### Assumption:
1. Since client closes connection unexpectedly (*Broken Pipe*) - this may not well implemented in `lwip`, e.g. it frees `tcp` structure, sets pointer to `NULL`, but doesn't fire *"broken pipe"* event (return code, set flag somewhere, whatever)
2. As a next step you call `poll()` and hit `NULL` pointer -> **crash**.

I'd be happy to know better solution to fix this annoying crash. Unfortunately, I don't have bunch of time today to investigate / figure out problem deeply.